### DISCUSSION TITLE
Delete few static cache pages on saving/updating entry

### DISCUSSION THREAD

**Discussion by imakeitworknet** [statamic/cms#8573] [Q&A] [answered]
Hi.
I'm using full static cache in my site (Statamic 3.4.11) and I want to make a small modification.

I need a function which would trigger removing a few files (pages) in static cache folder when user creates a new entry or edits an entry.

For example when user adds an entry to Collection NEWS (hits Publish or Update), function needs to run a function like:

IF entry belongs to Collection NEWS, delete static cache pages X,Y,Z.
If entry belongs to Collection EVENTS, delete static cache pages M and X.

I know the file names, but I'm not sure where I could trigger that function (and also get Collection name WHEN SAVING OR UPDATING entry).

Thanks for ideas!

**Comment by duncanmcclean** [comment, marked as answer]
You could create a listener that listens for the `EntryCreated`/`EntrySaved` events, then do your custom logic in the listener.

We do something similar with a listener: https://steadfastcollective.com/articles/handling-statamic-static-cache-invalidation-on-large-sites

**Reply by imakeitworknet** [reply]
That's great, exactly what I was looking for. Thanks!
